Extension/src/LanguageServer/settings.ts

```diff
@@ -996,28 +860,71 @@ export class OtherSettings {
         this.resource = resource;
     }
 
-    public get editorTabSize(): number | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<number>("tabSize"); }
-    public get editorInsertSpaces(): boolean | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<boolean>("insertSpaces"); }
-    public get editorAutoClosingBrackets(): string | undefined { return vscode.workspace.getConfiguration("editor", this.resource).get<string>("autoClosingBrackets"); }
-    public get filesEncoding(): string | undefined { return vscode.workspace.getConfiguration("files", { uri: this.resource, languageId: "cpp" }).get<string>("encoding"); }
-    public get filesAssociations(): any { return vscode.workspace.getConfiguration("files").get("associations"); }
-    public set filesAssociations(value: any) {
-        void vscode.workspace.getConfiguration("files").update("associations", value, vscode.ConfigurationTarget.Workspace);
+    private getAsString(sectionName: string, settingName: string, resource: any, defaultString: string): string {
+        const fullConfiguration = vscode.workspace.getConfiguration(sectionName, resource);
+        const value = fullConfiguration.get<string>(settingName);
+
```

Comment:
The two blank lines surrounding the `if (isString)` don't occur in the other methods.